PlaceRouteHierFlow/placer/ILP_solver.cpp

```diff
@@ -358,7 +358,9 @@ double ILP_solver::GenerateValidSolution(design& mydesign, SeqPair& curr_sp, PnR
 
   // set_add_rowmode(lp, FALSE);
   {
-    double row[N_var + 1] = {0};
+
+    std::vector<double> row( N_var+3, 0);
```

Comment:
@854768750 @srini229 @Lastdayends Seems that we need at least 2 extra elements in the array to not have out of bounds errors. Can someone figure out what the correct number should be?

Reply:
But I got no "out of bounds" error. Which command did you use?

Reply:
I used:
```
python -m pytest -s -k high_speed_comparator -- tests/test_cmdline.py
```

Reply:
But of course to get the error you need to change the size of row back to N_var+1.
The question is whether `N_var+3` is the right number always, or just for this example.

Reply:
Thanks @stevenmburns . It should be `N_var+1`. The actual problem is from redundant elements in seqpair.
For example, the hierarchy "ccn" has one block, but its negpair has two elements (one is redundant). 
I added one line to skip this redundant elements to avoid out of bound.

Reply:
@srini229 I found that in hierarchy `ccn` of `high_speed_comparator`, the pospair and negpair both have 2 elements (but there is only 1 block). Do you know why there are two elements here?
![picture](https://user-images.githubusercontent.com/28666144/129406112-3d1fa75c-c208-4d9e-ab56-2b7a8ff57ab1.PNG)

Reply:
The below line is adding an extra element for every symmetry constraint. This might be redundant in this case since the hierarchy has only one block.

[  posPair.push_back(bit->dnode); // axis --> positive ](https://github.com/ALIGN-analoglayout/ALIGN-public/blob/26e23501c0da1f569fdf837f75098382139100a0/PlaceRouteHierFlow/placer/SeqPair.cpp#L357
)

Reply:
ok